R/get_v1_data.R

```diff
@@ -209,35 +212,157 @@ spod_clean_zones_v1 <- function(zones_path) {
+#'   \item{viajes}{\code{double}. The number of trips taken within the specified time period and distance.}
+#'   \item{viajes_km}{\code{double}. The total length of all trips in kilometers for the specified time period and distance.}
+#'   \item{day}{\code{double}. The day of the trip.}
+#'   \item{month}{\code{double}. The month of the trip.}
+#'   \item{year}{\code{double}. The year of the trip.}
+#' }
+#' 
+#' @export 
+#' @examples
+#' \dontrun{
+#' 
+#' # create a connection to the v1 data
+#' Sys.setenv(SPANISH_OD_DATA_DIR = "~/home/nosync/cache/mitma")
+#' dates <- c("2020-02-14", "2020-03-14", "2021-02-14", "2021-02-14", "2021-02-15")
+#' od_dist <- spod_get_od_v1(zones = "distr", dates = dates)
+#' 
+#' # od dist is a table view filtered to the specified dates
+#' 
+#' # access the source connection with all dates
+#' # list tables
+#' DBI::dbListTables(od_dist$src$con)
+#' }
+spod_get_od_v1 <- function(
+  zones = c("districts", "dist", "distr", "distritos",
+    "municipalities", "muni", "municip", "municipios"),
   dates = NULL,
   data_dir = spod_get_data_dir(),
   quiet = FALSE,
-  read_fun = duckdb::tbl_file
+  duck_max_mem = 2,
+  duck_max_threads = parallelly::availableCores()
 ) {
-  # Processing of the date arguments is performed in `spod_download_data()`
+  # hardcode od as this is a wrapper to get origin-destiation data
+  type <- "od"
   
   zones <- match.arg(zones)
+  zones <- spod_zone_names_en2es(zones)
+  
+  
+  if (is.character(dates)) {
+    if ( dates != "cached" ) {
+      dates <- spod_dates_argument_to_dates_seq(dates = dates)
+      # use the spot_download_data() function to download any missing data
+    spod_download_data(
+      type = type,
+      zones = zones,
+      dates = dates,
+      data_dir = data_dir,
+      return_output = FALSE
+    )
+    }
+  }
+
+  # create in memory duckdb connection
+  drv <- duckdb::duckdb()
+  con <- DBI::dbConnect(drv, dbdir = ":memory:", read_only = FALSE)
+
+  # define memory and threads limits
+  con <- spod_duckdb_limit_resources(
+    con = con,
+    duck_max_mem = duck_max_mem,
+    duck_max_threads = duck_max_threads
+  )
   
-  # use the spot_download_data() function to download any missing data
-  downloaded_files <- spod_download_data(
-    type = "od",
+
+  # attach the od folder of csv.gz files with predefined and cleaned up data types
+  con <- spod_duckdb_od_v1(
+    con = con,
     zones = zones,
-    dates = dates,
     data_dir = data_dir
   )
+  
+  # filter by date
+  # actually, it seems like this works even if we do not return the 'con' from the function below, but I guess it is safer to return the 'con' and resave it to the 'con' of the environment/scope of this function
+  if (is.character(dates)) {
+    if ( dates != "cached" ) {
+      con <- spod_duckdb_filter_by_dates(con, "od_csv_clean", "od_csv_clean_filtered", dates)
+    }
+  }
+  
 
-
-  # read data from cached files
+  # DBI::dbListTables(con) # for debugging only
```

Comment:
This should probably moves somewhere else, e.g. `data-raw/ad-hoc-tests`.

Reply:
Will cleanup, but for debugging while stepping through the function manually this is immensely convenient. There are plenty more comments and commented code to remove.